Sources/FlushRequest.swift

```diff
@@ -39,14 +38,19 @@ class FlushRequest: Network {
                                              queryItems: [URLQueryItem(name: "ip", value: ipString)],
                                              headers: ["Content-Type": "application/json"],
                                              parse: responseParser)
-
+        var result = false
+        let semaphore = DispatchSemaphore(value: 0)
         flushRequestHandler(BasePath.getServerURL(identifier: basePathIdentifier),
                             resource: resource,
                             completion: { success in
-                                completion(success)
+                                result = success
+                                semaphore.signal()
         })
+        _ = semaphore.wait(timeout: DispatchTime.distantFuture)
```

Comment:
Maybe, we should change this timeout to some number of minutes?

Reply:
yeah, agreed.